service/assessment/assessment.go

```diff
@@ -479,7 +493,9 @@
 	}
 
 	// Retrieve our cached entry
+	s.confMutex.Lock()
```

Comment:
Its a judgement call. In general, this is done by two different mutex structs. `Mutex` only knows `Lock`, which means that this mutex is locked for reading and writing (by all and from all). This is suitable for most scenarios.

There is also the more specialized variant `RWMutex` which allows to separate locking for reading and writing. The idea is that multiple readers can `RLock` the variable in parallel (which is not possible with the regular `Lock`). It will however prevent any write in the meantime. And the other way around, a regular (write) `Lock` will prevent any readers to access it while the writing lock is active. This is especially suited for scenarios where you have far more readers than writers and you want to enable concurrent reading.